src/Widgets/FadeBin.vala

```diff
@@ -0,0 +1,150 @@
+public class Tuba.Widgets.FadeBin : Gtk.Widget {
+	const int MAX_HEIGHT = 300;
+	const float FADE_HEIGHT = 125f;
+
+	private Gtk.Widget? _child = null;
+	public Gtk.Widget? child {
+		get { return _child; }
+		set {
+			if (_child != null) _child.unparent ();
+			_child = value;
+			_child.set_parent (this);
+		}
+	}
+
+	~FadeBin () {
+		if (this.child != null) this.child.unparent ();
```

Comment:
This should be done in [`dispose`](https://valadoc.org/gobject-2.0/GLib.Object.dispose.html), something like this:

```vala
public override void dispose () {
    this.child = null;
    base.dispose ();
}
```